preprocess_normalize.py

```diff
@@ -132,49 +119,38 @@ def generate_centerline(dataset_info, regenerate = False, algo_fitting = 'linear
-            else:
-                print(subject_name + ' SC segmentation does not exist. Extracting centerline from ' + fname_image)
-                im_seg = Image(fname_image).change_orientation('RPI')
-                param_centerline = ParamCenterline(algo_fitting = 'optic', smooth = smooth, degree = 5, minmax = minmax)
-
-            # extracting intervertebral discs
-            im_discs = Image(fname_image_discs).change_orientation('RPI')
-            coord = im_discs.getNonZeroCoordinates(sorting = 'z', reverse_coord = True)
-            coord_physical = []
-            for c in coord:
-                if c.value <= last_disc or c.value in [48, 49, 50, 51, 52]:
-                    c_p = list(im_discs.transfo_pix2phys([[c.x, c.y, c.z]])[0])
-                    c_p.append(c.value)
-                    coord_physical.append(c_p)
-
-            # extracting centerline
-            im_ctl, arr_ctl, arr_ctl_der, _ = get_centerline(im_seg, param = param_centerline, space = 'phys')
-
-            # save centerline as .nii.gz file
-            im_ctl.save(fname_centerline + '.nii.gz', dtype = 'float32')
-            centerline = Centerline(points_x = arr_ctl[0], points_y = arr_ctl[1], points_z = arr_ctl[2], deriv_x = arr_ctl_der[0], deriv_y = arr_ctl_der[1], deriv_z = arr_ctl_der[2])
-            centerline.compute_vertebral_distribution(coord_physical)
-            # save centerline .npz file
-            centerline.save_centerline(fname_output = fname_centerline)
+            print(subject_name + ' SC segmentation does not exist. Extracting centerline from ' + fname_image)
+            im_seg = Image(fname_image).change_orientation('RPI')
+            param_centerline = ParamCenterline(algo_fitting = 'optic', smooth = smooth, degree = 5, minmax = minmax)
+
+        # extracting intervertebral discs
+        im_discs = Image(fname_image_discs).change_orientation('RPI')
+        coord = im_discs.getNonZeroCoordinates(sorting = 'z', reverse_coord = True)
+        coord_physical = []
+        for c in coord:
+            if c.value <= last_disc or c.value in [48, 49, 50, 51, 52]:
+                c_p = list(im_discs.transfo_pix2phys([[c.x, c.y, c.z]])[0])
+                c_p.append(c.value)
+                coord_physical.append(c_p)
+
+        # extracting centerline
+        _, arr_ctl, arr_ctl_der, _ = get_centerline(im_seg, param = param_centerline, space = 'phys')
+        centerline = Centerline(points_x = arr_ctl[0], points_y = arr_ctl[1], points_z = arr_ctl[2], deriv_x = arr_ctl_der[0], deriv_y = arr_ctl_der[1], deriv_z = arr_ctl_der[2])
+        centerline.compute_vertebral_distribution(coord_physical)
 
         list_centerline.append(centerline)
         tqdm_bar.update(1)
     tqdm_bar.close()
     os.chdir(current_path)
     return list_centerline
 
-# def compute_ICBM152_centerline(dataset_info): ###### FIX
+# def compute_ICBM152_centerline(dataset_info): # ??????
```

Comment:
replace the ????? by an issue on GH that points to the line of code that you don't understand